src/videojs-contrib-hls.js

```diff
@@ -139,13 +139,37 @@ export default class HlsHandler extends Component {
         this.options_[option] = this.source_[option];
       }
     });
-    this.masterPlaylistController_ = new MasterPlaylistController({
-      url: this.source_.src,
-      withCredentials: this.options_.withCredentials,
-      currentTimeFunc: this.tech_.currentTime.bind(this.tech_),
-      mediaSourceMode: this.options_.mode,
-      hlsHandler: this,
-      externHls: Hls
+    this.options_.url = this.source_.src;
+    this.options_.tech = this.tech_;
+    this.options_.externHls = Hls;
+    this.options_.bandwidth = this.bandwidth;
+    this.masterPlaylistController_ = new MasterPlaylistController(this.options_);
```

Comment:
It's minor, but passing the whole this.options_ object when creating MasterPlaylistController feels a bit strange to me. Maybe it all is needed right now for MasterPlaylistController, and in the future, if we're ever leaking more information than we should, we can just specify what we need to send then. But it just feels a bit odd.